app/controllers/search_controller.rb

```diff
@@ -28,8 +45,24 @@ def chat_channel_params
       channel_status
       status
     ]
+
+    params.permit(accessible)
+  end
+
+  def classified_listing_params
+    remove_blank_params
+    params.permit(CLASSIFIED_LISTINGS_PARAMS)
+  end
+
+  def format_integer_params
     params[:page] = params[:page].to_i if params[:page].present?
     params[:per_page] = params[:per_page].to_i if params[:per_page].present?
-    params.permit(accessible)
+  end
+
+  # Some Elasticsearches/QueryBuilders treat values such as empty Strings and
+  # nil differently. This is a helper method to remove any params that are
+  # blank before passing it to Elasticsearch.
+  def remove_blank_params
+    params.delete_if { |_k, v| v.blank? }
```

Comment:
After a great convo with @vaidehijoshi I've changed this up a little bit. The challenge from a development perspective is that each `QueryBuilder` for each Model in Elasticsearch handles params a little differently. For example, some `QueryBuilders` will clean up `nil` or empty String values automatically. Others might let it through (a behavior we'd like to intentionally keep right now). When making a request from the frontend, a dev basically has to go through a lot to understand what to pass in and how it's treated.

Instead, I'm starting a new trend where we keep general params cleanup here in the `search_controller` with the idea that in the future a dev would only have to check the list of whitelisted params and should be able to get most of the way without having to dig further into the various `Search` modules, including the `QueryBuilder` to understand how things like `""` are being handled.

The controller is also easy to implement clean up across all searches or specific ones vs changing code in _each_ `QueryBuilder` for a similar, all-encompassing change. Eventually, I suspect we'll refactor `QueryBuilders`, but that's further off right now.